### PR TITLE
Workbox extending

### PR DESCRIPTION
**Pull request by gauntface**
This adds some review comments from Surma and adds docs for:

- workbox-cli
- workbox-cachable-response
- workbox-background-sync
- workbox-google-analytics

### CODE REVIEW

File: src/content/en/tools/workbox/v3/_index.yaml [hunk 1]
```diff
@@ -21,8 +21,8 @@ landing_page:
         Getting started with Workbox is easy, in a few minutes you'll be
         caching assets for your web site and serving them up in your web app.
       custom_image:
-        background: google-green
-        icon_name: history
+        background: orange
+        icon_name: computer
```
Comment (addyosmani) [inline src/content/en/tools/workbox/v3/_index.yaml]: Without being able to view this image, I'm notionally supportive of a switch to iconography that isn't just history :) Confirmed this change isn't staged anywhere in chat. Should be fine however.

File: src/content/en/tools/workbox/v3/modules/workbox-cacheable-response.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
+project_path: /web/tools/workbox/v3/_project.yaml
+book_path: /web/tools/workbox/v3/_book.yaml
+description: The module guide for workbox-cacheable-response.
+
+{# wf_updated_on: 2017-11-27 #}
+{# wf_published_on: 2017-11-27 #}
+
+# Workbox Cachable Response  {: .page-title }
```
Comment (addyosmani) [inline src/content/en/tools/workbox/v3/modules/workbox-cacheable-response.md]: The title of this entry is Workbox Cachable Response but the description uses 'Cacheable'. Typo for the earlier one?

File: src/content/en/tools/workbox/v3/modules/workbox-cacheable-response.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+whether a response should be cached based on its
+[numeric status code](https://developer.mozilla.org/en-US/docs/Web/API/Response/status),
+the presence of a
+[header](https://developer.mozilla.org/en-US/docs/Web/API/Response/headers)
+with a specific value, or a combination of the two.
+
+## Caching Based on Status Codes
+
+You can configure a [Workbox strategy](./workbox-strategies) to consider
+set of status codes as being eligible for caching by setting the `cacheableResponse` option with an array of `statuses`:
```
Comment (addyosmani) [inline src/content/en/tools/workbox/v3/modules/workbox-cacheable-response.md]: to consider a set of status codes

**Comment by PaulKinlan** [comment]
I'd suggest Blink>ServiceWorker as the component.

File: src/content/en/tools/workbox/v3/modules/workbox-cli.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+process, with flexible configurations.
+
+## Install the CLI
+
+Installing the CLI is simple, just run the following command in your
+terminal.
+
+* {NPM}
+
+    ```
```
Comment (petele) [inline src/content/en/tools/workbox/v3/modules/workbox-cli.md]: Check out this [syntax for terminal code](https://developers.google.com/internal/publishing/reference/styles#terminal_code)

Also, should it be indented when you're wrapping in triple back ticks?

File: src/content/en/tools/workbox/v3/modules/workbox-cli.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+`generateSW` mode.
+
+Most developers will only need to run workbox wizard once, and you're free
+to manually customize the initial configuration file that's generated,
+using any of the supported build configuration options.
+
+To start the wizard run:
+
+```
+workbox wizard
```
Comment (petele) [inline src/content/en/tools/workbox/v3/modules/workbox-cli.md]: Check out this [syntax for terminal code](https://developers.google.com/internal/publishing/reference/styles#terminal_code)

File: src/content/en/tools/workbox/v3/modules/workbox-google-analytics.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+Google Analytics does not have a built-in dimension for online vs. offline
+interactions. However, you can create your own dimension for exactly this
+purpose using a feature called
+[custom dimensions](https://support.google.com/analytics/answer/2709828).
+
+To track requests that were replayed by the service worker using a custom
+dimension with Workbox Google Analytics, follow these steps:
+
+**1)** [Create a new custom dimension](https://support.google.com/analytics/answer/2709829)
```
Comment (petele) [inline src/content/en/tools/workbox/v3/modules/workbox-google-analytics.md]: Should these be true ordered lists?

Comment (gauntface) [reply src/content/en/tools/workbox/v3/modules/workbox-google-analytics.md]: The formatting with the indented code block breaks.

Comment (petele) [reply src/content/en/tools/workbox/v3/modules/workbox-google-analytics.md]: There is a way to do it, I think you have to indent 4 or 8 spaces?

Comment (petele) [reply src/content/en/tools/workbox/v3/modules/workbox-google-analytics.md]: i think it's https://stackoverflow.com/questions/18088955/markdown-continue-numbered-list

Comment (gauntface) [reply src/content/en/tools/workbox/v3/modules/workbox-google-analytics.md]: It doesn't work but I've done it and lets see what happens on prod.

**Pull request merged**